### ISSUE TITLE
FSharpSynchronousIndentationService is, well, UI synchronous

### ISSUE THREAD

**Issue by dsyme** [dotnet/fsharp#10156]
I'm debugging UI freeze ups when hitting "Return" in scripts and pasting into some scripts. 

This is a devenv.exe UI callstack featuring "FSharpSynchronousIndentationService.GetIndentation" which is resulting in a synchronous reactor call

I'll look into fixing it

```
>	FSharp.Core.dll!Microsoft.FSharp.Control.AsyncPrimitives.ResultCell<Microsoft.FSharp.Control.AsyncResult<Microsoft.CodeAnalysis.ExternalAccess.FSharp.Editor.FSharpIndentationResult?>>.TryWaitForResultSynchronously(Microsoft.FSharp.Core.FSharpOption<int> timeout) Line 792	F#
 	FSharp.Core.dll!Microsoft.FSharp.Control.FSharpAsync.RunSynchronously<Microsoft.CodeAnalysis.ExternalAccess.FSharp.Editor.FSharpIndentationResult?>(Microsoft.FSharp.Control.FSharpAsync<Microsoft.CodeAnalysis.ExternalAccess.FSharp.Editor.FSharpIndentationResult?> computation, Microsoft.FSharp.Core.FSharpOption<int> timeout, Microsoft.FSharp.Core.FSharpOption<System.Threading.CancellationToken> cancellationToken) Line 1139	F#
 	Microsoft.CodeAnalysis.ExternalAccess.FSharp.dll!Microsoft.CodeAnalysis.ExternalAccess.FSharp.Internal.Editor.FSharpSynchronousIndentationService.GetIndentation(Microsoft.CodeAnalysis.Document document, int lineNumber, Microsoft.CodeAnalysis.Formatting.FormattingOptions.IndentStyle indentStyle, System.Threading.CancellationToken cancellationToken)	Unknown
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Indentation.IIndentationServiceExtensions.GetIndentation(Microsoft.CodeAnalysis.Indentation.IIndentationService service, Microsoft.CodeAnalysis.Document document, int lineNumber, System.Threading.CancellationToken cancellationToken)	Unknown
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.SmartIndent.SmartIndent.GetDesiredIndentation(Microsoft.VisualStudio.Text.ITextSnapshotLine lineToBeIndented, System.Threading.CancellationToken cancellationToken)	Unknown
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.SmartIndent.SmartIndent.GetDesiredIndentation(Microsoft.VisualStudio.Text.ITextSnapshotLine line)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.SmartIndentationService.GetDesiredIndentation(Microsoft.VisualStudio.Text.Editor.ITextView textView, Microsoft.VisualStudio.Text.ITextSnapshotLine line)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Operations.Implementation.EditorOperations.PositionCaretWithSmartIndent(Microsoft.VisualStudio.Text.ISelectionTransformer transformer, bool useOnlyVirtualSpace, bool extendSelection)	Unknown
 	Microsoft.VisualStudio.Platform.VSEditor.dll!Microsoft.VisualStudio.Text.Operations.Implementation.EditorOperations.InsertNewLine.AnonymousMethod__1(Microsoft.VisualStudio.Text.ISelectionTransformer transformer)	Unknown
```

**Comment by dsyme** [comment]
This is fixed in #10159 (the operation is still synchronous, but doesn't invoke the reactor thread)

**Comment by cartermp** [comment]
Will close this out as a part of #10159 even though the synchronous nature of this isn't changed.
